ts/features/fci/screens/valid/FciDocumentPreviewScreen.tsx

```diff
@@ -1,16 +1,152 @@
+import { pipe } from "fp-ts/lib/function";
 import React from "react";
+import { SafeAreaView, StyleSheet } from "react-native";
+import * as O from "fp-ts/lib/Option";
+import * as S from "fp-ts/lib/string";
+import { IOColors } from "../../../../components/core/variables/IOColors";
+import BaseScreenComponent from "../../../../components/screens/BaseScreenComponent";
+import TouchableDefaultOpacity from "../../../../components/TouchableDefaultOpacity";
+import IconFont from "../../../../components/ui/IconFont";
+import I18n from "../../../../i18n";
 import { IOStackNavigationRouteProps } from "../../../../navigation/params/AppParamsList";
+import { emptyContextualHelp } from "../../../../utils/emptyContextualHelp";
 import { DocumentViewer } from "../../components/DocumentViewer";
 import { FciParamsList } from "../../navigation/params";
+import { useIODispatch, useIOSelector } from "../../../../store/hooks";
+import { fciSignatureDetailDocumentsSelector } from "../../store/reducers/fciSignatureRequest";
+import DocumentsNavigationBar from "../../components/DocumentsNavigationBar";
+import {
+  fciDownloadPreviewClear,
+  fciShowSignedDocumentsEndRequest
+} from "../../store/actions";
+import { fciDownloadPathSelector } from "../../store/reducers/fciDownloadPreview";
 
 export type FciDocumentPreviewScreenNavigationParams = Readonly<{
-  documentUrl: string;
+  documentUrl?: string;
 }>;
 
+const styles = StyleSheet.create({
+  container: {
+    flex: 1
+  }
+});
+
 export const FciDocumentPreviewScreen = (
   props: IOStackNavigationRouteProps<FciParamsList, "FCI_DOC_PREVIEW">
 ): React.ReactElement => {
-  const documentUrl = props.route.params.documentUrl;
+  const [documentUrl, setDocumentUrl] = React.useState("");
+  const [showDocNavBar, setShowDocNavBar] = React.useState(false);
+  const [totalPages, setTotalPages] = React.useState(0);
+  const [currentPage, setCurrentPage] = React.useState(0);
+  const [currentDoc, setCurrentDoc] = React.useState(0);
+  const documents = useIOSelector(fciSignatureDetailDocumentsSelector);
+  const fciDownloadPath = useIOSelector(fciDownloadPathSelector);
+  const dispatch = useIODispatch();
+
+  React.useEffect(() => {
+    const docUrl = pipe(
+      props.route.params,
+      O.fromNullable,
+      O.chainNullableK(p => p.documentUrl),
+      O.getOrElse(() => "")
+    );
+
+    if (docUrl.length > 0) {
+      setDocumentUrl(docUrl);
+    } else if (documents.length > 0) {
+      setShowDocNavBar(true);
+      setDocumentUrl(documents[currentDoc].url);
+    }
+  }, [currentDoc, documents, props.route.params]);
```

Comment:
What you think of narrowing down the dependencies from all the params to only documentUrl?

Reply:
What about [currentDoc, document, props.route.params.documentUrl]? Can it work?

Reply:
params could be undefined because `documentUrl` is optional if I try to add in the dependencies array `props.route.params.documentUrl` I have a runtime error when params is `undefined`.

Reply:
@francescopersico maybe a solution like these could works

```typescript 
const [documentUrl, setDocumentUrl] = React.useState(
    props.route.params?.documentUrl ?? ""
  );
 // ...
  React.useEffect(() => {
    if (documents.length > 0 && S.isEmpty(documentUrl) === true) {
      setShowDocNavBar(true);
      setDocumentUrl(documents[currentDoc].url);
    } else {
      setShowDocNavBar(false);
    }
  }, [currentDoc, documents, documentUrl]);
```

Reply:
@francescopersico fixed passing right params during navigation